core/foundation/test/testClassEdit.cxx

```diff
@@ -292,3 +292,20 @@ TEST(TClassEdit, GetNormalizedName)
    TClassEdit::GetNormalizedName(n, "std::vector<float, class std::allocator<float>>");
    EXPECT_STREQ("vector<float>", n.c_str());
 }
+
+// https://github.com/root-project/root/issues/18654
+TEST(TClassEdit, UnorderedMapNameNormalization)
+{
+   // These two should normalise to map<string,char>.
+   // When this did not work, df104_CSVDataSource-py crashed while querying the classes
+   std::string in_cxx11{"std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, char, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, char> > >"};
+   std::string in{"std::unordered_map<std::basic_string<char, std::char_traits<char>, std::allocator<char> >, char, std::hash<std::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::basic_string<char, std::char_traits<char>, std::allocator<char> > const, char> > >"};
+   const auto target = "unordered_map<string,char>";
+
+   std::string out;
+   TClassEdit::GetNormalizedName(out, in);
+   EXPECT_STREQ(target, out.c_str());
+
+   TClassEdit::GetNormalizedName(out, in_cxx11);
+   EXPECT_STREQ(target, out.c_str());
+}
```

Comment:
@pcanal @dpiparo here is the test for the normalised name, which passes when the interpreterHelper isn't instantiated yet, but fails when it is.

In the former case, we get
`unordered_map<string,char>`
and in the latter:
`unordered_map<string,char,hash<string>,equal_to<string>,allocator<pair<const string,char> > >`

What should be the correct result?

Reply:
`unordered_map<string,char>` is the normalized name.

Reply:
Right.  This does not return the normalized name but the type which indeed might/should still have the default template arguments.  If I remember correctly the are meant to be remove lexicographicaly in a higher level routine.

Reply:
I.e `GetNormalizedType` can/should return with the default parameter
but in the end `GetNormalizedName` needs to drop them (for STL collections) both compiled and interpreted code.

Reply:
> I.e `GetNormalizedType` can/should return with the default parameter
> but in the end `GetNormalizedName` needs to drop them (for STL collections) both compiled and interpreted code.

I will implement this now.